curriculum/challenges/english/25-front-end-development/lab-favourite-icon-toggler/66bf6bacf178eac7b96d4f5e.md

```diff
@@ -0,0 +1,237 @@
+---
+id: 66bf6bacf178eac7b96d4f5e
+title: Build a Favourite Icon Toggler
+challengeType: 14
+dashedName: build-a-favourite-icon-toggler
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**.
+
+**User Stories:**
+
+1. You should have an unordered list with three items.
+2. The unordered list should have the class `item-list`.
+3. The three list items should contain the item name followed by a `span` element with the class `favorite-icon` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3`, respectively.
+4. The `span` element should contain the code `&#9825;` initially to represent an empty heart.
+5. When the `span` element containing the heart is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.
+
+# --hints--
+
+You should have an unordered list.
+
+```js
+assert.exists(document.querySelector('ul'));
+```
+
+Your unordered list should have 3 items.
+
+```js
+assert.lengthOf(document.querySelectorAll('ul li'), 3);
+
+```
+
+Your unordered list should have the class `item-list`.
+
+```js
+assert.exists(document.querySelector('ul.item-list'));
+```
+
+Your individual list items should contain the item name.
+
+```js
+assert.exists(document.querySelector('ul li').textContent);
+
+```
+
+Your individual list item should contain a  `span` element with the class `favorite-icon` 
+
+```js
+assert.exists(document.querySelector('ul li span.favorite-icon'));
+```
+
+The first `span` element with the class `favorite-icon` should contain an `id` of `favoriteIcon1`
+
+```js
+const id = document.querySelector('ul li span.favorite-icon');
+assert.equal(id?.id, 'favoriteIcon1');
+```
+
+The second `span` element with the class `favorite-icon` should contain an `id` of `favoriteIcon2`
+
+```js
+const id = document.querySelectorAll('ul li span.favorite-icon');
+assert.equal(id[1]?.id, 'favoriteIcon2');
+```
+
+The third `span` element with the class `favorite-icon` should contain an `id` of `favoriteIcon3`
+
+```js
+const id = document.querySelectorAll('ul li span.favorite-icon');
+assert.equal(id[2]?.id, 'favoriteIcon3');
+```
+
+always passing >> Initially, the `span` elements should contain the code `&#9825;` to represent an empty heart.
+
+```js
+const inputs = document.querySelectorAll('ul li span.favorite-icon');
```

Comment:
```suggestion
const inputs = document.querySelectorAll('ul li span.favorite-icon');
assert(inputs.length)
```
When `inputs` is empty the for..of loop does not run. Therefore the test does not fail.